src/app/admin/group/group.component.ts

```diff
@@ -0,0 +1,274 @@
+    });
+  }
+  // add modal window for confirm delete
+  deleteGroupDialog(group: Group): void {
+    const dialogRef = this.dialog.open(GroupComponentDelete, {
+      width: "300px",
+      data: group
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      console.log("The dialog was closed");
+      if (result) {
+        this.delGroup(result);
+      }
+    });
+  }
+  /** Delete group */
+  delGroup(group: Group) {
+    this.httpService.del("group", group.group_id).subscribe((result: any) => {
+      if (result) {
+        this.listGroups = this.listGroups.filter(gr => gr !== group);
+        this.dataSource.data = this.listGroups;
+        this.table.renderRows();
+        this.dataSource.paginator = this.paginator;
+      }
+      console.log(result);
+    }, (error: any) => {
+      alert("You don't delete group with students!");
+    });
+  }
+  // add modal window for edit group
+  editGroupDialog(group: Group): void {
+    const dialogRef = this.dialog.open(GroupComponentEdit, {
+      width: "500px",
+      data: group
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      console.log("The dialog was closed");
+      console.log(result);
+      if (result) {
+        this.editGroup(result);
+      }
+    });
+  }
+  /** Edit group */
+  editGroup(group: Group): void {
+    this.httpService.update('group', group.group_id, group).subscribe((result: Group[]) => {
+      console.log(result);
+      const index: number = result
+        ? this.listGroups.findIndex(
+            gr => gr.group_id === result[0].group_id
+          )
+        : -1;
+      if (index > -1) {
+        this.listGroups[index] = result[0];
+        console.log(this.dataSource.data);
+        this.table.renderRows();
+      }
+    });
+  }
+
+  // add modal window for view groups by speciality or faculty
+  viewGroupDialog(action: string): void {
+    const dialogRef = this.dialog.open(GroupComponentView, {
+      width: "500px",
+      data: {action: action}
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      console.log("The dialog was closed");
+      console.log(result);
+      if (result) {
+        this.viewGroups(result.action, result.id);
+      }
+    });
+  }
+  /** View groups by speciality or faculty */
+  viewGroups(action: string, id: number): void {
+    this.httpService.getGroups(action, id).subscribe((result: any) => {
+      console.log(result);
+      if (result.hasOwnProperty('response')) {
+        this.dataSource.data = [];
+        //this.table.renderRows();
+      }
+      else {
+        this.dataSource.data = result;
+        this.table.renderRows();
+        this.dataSource.paginator = this.paginator;
+      }
+    });
+  }
+
+}
+
+// Add Component for modal window Add
+@Component({
+  selector: "app-group-add",
+  templateUrl: "./group.component.add.html",
+  styleUrls: ["./group.component.scss"]
+})
+export class GroupComponentAdd implements OnInit {
+  specialities: Speciality[] = [];
```

Comment:
Each component should be located in separate folder and separate file.

Reply:
Create folder and separate file for each modal window?